tools/update.sh

```diff
@@ -72,52 +62,35 @@ get_root_folder ()
     echo "${parent_folder}"
 }
 
-update_branch ()
-{
-    trimmed_branch_name=`echo "${OPT_BRANCH}" | sed 's/\///g'`
-    if [ "${trimmed_branch_name}" == "${OPT_BRANCH}" ]
-    then
-        # Not a remote branch, so update the local one.
-        echo "Updating ${OPT_BRANCH} from origin."
-        git merge origin/${OPT_BRANCH}
-    fi
-    git submodule init
-    git submodule sync
-    git submodule update --init --recursive
-}
-
 update_repo ()
 {
     local repo_dir=$1
     local git_repo_url=$2
+    local git_branch=${OPT_BRANCH}
 
     if [ ! -d "$repo_dir" ]
     then
-        echo "Cloning $git_repo_url into $repo_dir"
-        git clone ${git_repo_url} ${repo_dir}
-        cd ${repo_dir}
+        echo "Cloning $git_repo_url at $git_branch into $repo_dir"
+        git clone --branch $git_branch --single-branch --depth 1 $git_repo_url $repo_dir
     else
-        echo "Found repo at $repo_dir.  Fetching the latest"
-        cd ${repo_dir}
-        git fetch origin
+        echo "Found repo at $repo_dir.  Remove and run update.sh again."
+        exit 2
     fi
-
-    echo "Checking out the ${OPT_BRANCH} branch."
-    git checkout ${OPT_BRANCH}
-    update_branch
-    cd ${ROOT_FOLDER}
 }
 
 ROOT_FOLDER=$(get_root_folder)
 ANDROID_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-Android.git"
 ANDROID_SDK_FOLDER="SalesforceMobileSDK-Android"
-IOS_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid.git"
-IOS_SDK_FOLDER="SalesforceMobileSDK-iOS-Hybrid"
+IOS_HYBRID_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid.git"
+IOS_HYBRID_SDK_FOLDER="SalesforceMobileSDK-iOS-Hybrid"
+IOS_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-iOS.git"
```

Comment:
We are not getting submodules so we need to get SalesforceMobileSDK-iOS directly.
We assume here that the SalesforceMobileSDK-iOS-Hybrid in branch x uses SalesforceMobileSDK-iOS submodule at branch x also.